### DISCUSSION TITLE
Excluding entries of an specific category

### DISCUSSION THREAD

**Discussion by robsonsobral** [ExpressionEngine/ExpressionEngine#3314] [Feature Requests]
**Is your feature request related to a problem? Please describe.**
This part of docs puzzles me:

> Note: If you are using exclusion (category="not 3|4") and an entry is in a category that is not excluded, the entry will be returned even if it also belongs to an excluded category.

**Describe the solution you'd like**
I need a way to **really** exclude entries of a specific category, but I can't think of a way to declare it on the `category` parameter or to name a parameter for this.

Does anybody have an idea?

- [x] I am capable and would like to work on implementation of this feature if it is considered.

**Comment by intoeetive** [comment]
@robsonsobral just to make sure, does `category="not 3&5"` also not work for your case?

**Comment by robsonsobral** [comment]
Hi, @intoeetive . I'm sorry it too so long for me to reply.

**Please**, help me here. For example:

```
{exp:channel:entries channel="service" limit="10"
    category="not 141" parse="inward"
    disable="category_fields|custom_fields|member_data" status="not closed"
    paginate="inline" uncategorized_entries="no"
}
```

Give us the following query:

```
SELECT DISTINCT t.entry_id , exp_channels.channel_id , t.entry_date, t.sticky
FROM exp_channel_titles AS t
LEFT JOIN exp_channels ON t.channel_id = exp_channels.channel_id
INNER JOIN exp_category_posts ON t.entry_id = exp_category_posts.entry_id
INNER JOIN exp_categories ON exp_category_posts.cat_id = exp_categories.cat_id
WHERE t.entry_id != ''
  AND t.site_id IN ('2')
  AND t.entry_date < 1637515910 AND (t.expiration_date = 0 OR t.expiration_date > 1637515910)
  AND t.channel_id IN (1,9,21,22,40,48,56,65,75,84,113,124,135,146,157,168,179,190,201,212,223,234,245,256,267,278,289,300,311,322,333,344,355,366,377,388,399,410,421,432,443,454,465,476,487,498,509,520,531,548,551,557,568,579,590,601,612,623,634)
  AND exp_categories.cat_id != '141'
  AND t.status != 'closed'
```

This **isn't** excluding the entries on category 141, but the occurrences of this category on entries. The ones which have more than one category, are still shown.

Should'nt the query use `LEFT OUTER JOIN` or something like that?

**Comment by intoeetive** [comment]
Per my understanding, this is is correct. 

So like, you have categories "Bread" and "Meat". You want to exclude "Bread" entries from the output, and so you specify `category="not 1"`, however the entry "Sandwich" is still in the output, because it also has "Meat" category.

But that's just my personal opinion here

**Comment by robsonsobral** [comment]
May I suggest something like, I don't know... `category_rules` or `type_of_category_rules`?

```html
category_rules="inclusive" category_rules="exclusive"
```

Being `inclusive` the way the module currently work and `exclusive` the way I need it to work?

---

May a native English speaker give me a good suggestion?

**Comment by intoeetive** [comment]
I think that makes a good FR

@matthewjohns0n @TomJaeger what are your thoughts?

**Comment by TomJaeger** [comment]
Totally agree... Good addition. We should also start planning a larger category scope as a whole soon.

**Comment by robsonsobral** [comment]
Or... Hmmmnnn... `category:TYPE="141"`? Is there a class/module/model/stuff dealing with this kind of modifiers already?

Do the words "exclusive" and "inclusive" make sense, @TomJaeger ?

**Comment by robinsowell** [comment]
This came up again today, I'd definitely like to have some way of doing 'If I say no meat entries, I mean NO MEAT entries, regardless if they're also in soup, salad or... pizza'.

And now I want pizza.  With meat on it.  I should have used brussel sprouts in the example.
